agent/http.go

```diff
@@ -543,6 +543,19 @@ func (s *HTTPHandlers) wrap(handler endpoint, methods []string) http.HandlerFunc
 		} else {
 			err = s.checkWriteAccess(req)
 
+			// Give the user a hint that they might be doing something wrong if their GET request
+			// has a non-empty body, which can easily happen using curl's --data-urlencode
+			// if specifying request type via "--request GET" rather than "--get".
+			// See https://github.com/hashicorp/consul/issues/11471.
+			if req.Method == http.MethodGet {
+				if req.ContentLength > 0 {
+					httpLogger.Warn("GET request has a non-empty body, "+
+						"check whether parameters were accidentally placed in the body rather than in the query string",
```

Comment:
I think the log is too long for most terminals; how's this for brevity?
```suggestion
					httpLogger.Warn("GET request has a non-empty body that will be ignored",
```

Reply:
I like the brevity (and the addition of "that will be ignored").

However, I don't think it's obvious to a user that they probably meant to do something else (like put the parameters in the query string). And the message will already go beyond 80 or 120 characters. For example, this is 148 characters:
```
2022-02-22T12:26:32.237-0800 [WARN]  agent.http: GET request has a non-empty body that will be ignored: url=/v1/agent/services from=127.0.0.1:48852
```

Does the brevity actually help make a message display on a single terminal line? Or is there some other benefit (e.g., maybe there are other reasons why this might have happened other than accidentally putting stuff that belongs in the query string into the request body)?

Reply:
Maybe we can include "that will be ignored" and keep the prescriptive part of the log too

Reply:
In that case, what do you think about the revised message?
```
GET request has a non-empty body that will be ignored; check whether parameters meant for the query string were accidentally placed in the body